### PR TITLE
added batch writes in Activity and Option ViewModels

### PR DESCRIPTION
**Pull request by j-cse**

### CODE REVIEW

**Comment by j-cse** [comment]
Whoops ran unit tests on another branch before pushing. Will fix batch writes

**Comment by j-cse** [comment]
Not sure how to handle compiler warning for this on line 65

> Consider using asynchronous alternative function

```swift
batch.commit { error in
            if let error = error {
                print("Error writing batch \(error)")
            } else {
                print("Batch write succeeded.")
            }
        }
```

**Comment by ApolloZhu** [comment]
You should be able to write

```swift
//   V wait for commit to finish before return
try await batch.commit()
//^ allow proper error reporting
```

File: Shared/ViewModel/ActivityViewModel.swift [hunk 1]
```diff
@@ -60,8 +61,14 @@ class ActivityViewModel: ObservableObject {
...
+        batch.commit { error in
+            if let error = error {
+                print("Error writing batch \(error)")
+            } else {
+                print("Batch write succeeded.")
+            }
+        }
         for member in allMembers {
             try await UserViewModel._addActivity(id: activityID, toUser: member)
         }
```
Comment (ApolloZhu) [inline Shared/ViewModel/ActivityViewModel.swift:74]: It's really this part that needs to be included in the batch commit

File: Shared/ViewModel/OptionViewModel.swift [hunk 2]
```diff
@@ -48,13 +48,25 @@ class OptionViewModel: ObservableObject {
...
+            try batch.setData(from: Option(id: optionID, title: title), forDocument: ref)
         }
+        
+        batch.commit { error in
+            if let error = error {
+                print("Error writing batch \(error)")
+            } else {
+                print("Batch write succeeded.")
+            }
+        }
```
Comment (ApolloZhu) [inline Shared/ViewModel/OptionViewModel.swift:68]: Should use try await

**Review by ApolloZhu — CHANGES_REQUESTED** [review]

**Pull request merged**